### PR TITLE
Issues/#160 SHA-3 conversion of Base58Check

### PR DESCRIPTION
**Pull request by macterra**

### CODE REVIEW

File: src/test/rpc_tests.cpp [hunk 1]
```diff
@@ -108,6 +108,7 @@ BOOST_AUTO_TEST_CASE(rpc_togglenetwork)
 
 BOOST_AUTO_TEST_CASE(rpc_rawsign)
 {
+#ifdef BUILD_BTC
```
Comment (macterra) [inline src/test/rpc_tests.cpp:111]: Opened #166

File: src/test/key_tests.cpp [hunk 2]
```diff
@@ -149,19 +173,23 @@ BOOST_AUTO_TEST_CASE(key_test1)
     BOOST_CHECK(key1.Sign(hashMsg, detsig));
     BOOST_CHECK(key1C.Sign(hashMsg, detsigc));
     BOOST_CHECK(detsig == detsigc);
-    BOOST_CHECK(detsig == ParseHex("304402205dbbddda71772d95ce91cd2d14b592cfbc1dd0aabd6a394b6c2d377bbe59d31d022014ddda21494a4e221f0824f0b8b924c43fa43c0ad57dccdaa11f81a6bd4582f6"));
+    // EQB_TODO generate new test data
+    // BOOST_CHECK(detsig == ParseHex("304402205dbbddda71772d95ce91cd2d14b592cfbc1dd0aabd6a394b6c2d377bbe59d31d022014ddda21494a4e221f0824f0b8b924c43fa43c0ad57dccdaa11f81a6bd4582f6"));
```
Comment (macterra) [inline src/test/key_tests.cpp:176]: Opened #167

**Review by ealefad — APPROVED** [review]

File: src/script/sigcache.cpp [hunk 3]
```diff
@@ -27,7 +27,6 @@ class CSignatureCache
 private:
      //! Entries are SHA256(nonce || signature hash || public key || signature):
     uint256 nonce;
-    // EQB_TODO: Update CuckooCache class to support SHA3
```
Comment (macterra) [inline src/script/sigcache.cpp:30]: Removed because CuckooCache supports SHA3 already indirectly

File: src/test/script_tests.cpp [hunk 4]
```diff
@@ -814,7 +814,11 @@ BOOST_AUTO_TEST_CASE(script_build)
     {
         CScript witscript = CScript() << ToByteVector(keys.pubkey0);
         uint256 hash;
+#ifdef BUILD_BTC
         CSHA256().Write(&witscript[0], witscript.size()).Finalize(hash.begin());
+#else  // BUILD_EQB
+        SHA3().Write(&witscript[0], witscript.size()).Finalize(hash.begin());
```
Comment (macterra) [inline src/test/script_tests.cpp:820]: This was an oversight unrelated to Base58Check

File: src/validation.cpp [hunk 5]
```diff
@@ -1391,7 +1391,11 @@ bool CheckInputs(const CTransaction& tx, CValidationState &state, const CCoinsVi
             // We only use the first 19 bytes of nonce to avoid a second SHA
             // round - giving us 19 + 32 + 4 = 55 bytes (+ 8 + 1 = 64)
             static_assert(55 - sizeof(flags) - 32 >= 128/8, "Want at least 128 bits of nonce for script execution cache");
+#ifdef BUILD_BTC
             CSHA256().Write(scriptExecutionCacheNonce.begin(), 55 - sizeof(flags) - 32).Write(tx.GetWitnessHash().begin(), 32).Write((unsigned char*)&flags, sizeof(flags)).Finalize(hashCacheEntry.begin());
+#else  // BUILD_EQB
+            SHA3().Write(scriptExecutionCacheNonce.begin(), 55 - sizeof(flags) - 32).Write(tx.GetWitnessHash().begin(), 32).Write((unsigned char*)&flags, sizeof(flags)).Finalize(hashCacheEntry.begin());
```
Comment (macterra) [inline src/validation.cpp:1397]: This was an oversight unrelated to Base58Check

**Review by r-soltani — APPROVED** [review]

**Pull request merged**